framework/include/base/MooseVariable.h

```diff
@@ -196,8 +196,16 @@ class MooseVariable : public MooseVariableBase
   const Node *& node() { return _node; }
   dof_id_type & nodalDofIndex() { return _nodal_dof_index; }
   bool isNodalDefined() { return _is_defined; }
-  const VariableValue & nodalSln() { return _nodal_u; }
-  const VariableValue & nodalSlnOld() { return _nodal_u_old; }
+  const VariableValue & nodalSln()
+  {
+    _need_nodal_u = true;
+    return _nodal_u;
+  }
+  const VariableValue & nodalSlnOld()
+  {
+    _need_nodal_u_old = true;
+    return _nodal_u_old;
+  }
```

Comment:
Yes, I do really care. Left you there a comment. You are using the wrong method there. You may want to re-read [my comment](https://github.com/idaholab/moose/issues/9690#issuecomment-323072876) on your #9690.